src/System.Private.Xml/src/System/Xml/Serialization/XmlSerializer.cs

```diff
@@ -121,14 +121,22 @@ public abstract class XmlSerializerImplementation
     /// </devdoc>
     public class XmlSerializer
     {
+#if uapaot
```

Comment:
I think it's fine to have the ifdef. Just because we need one hole in the dam doesn't mean we have to make it bigger than necessary.